src/playback/internal/au3/au3player.cpp

```diff
@@ -200,10 +214,17 @@ int Au3Player::playTracks(TrackList& trackList, double t0, double t1, const Play
         mixerLimit = t1;
     }
 
+    m_startOffset = options.startOffset;
+
     AudacityProject& project = projectRef();
     AudioIOStartStreamOptions sopts = ProjectAudioIO::GetDefaultOptions(project, true /*newDefault*/);
 
-    return audioEngine()->startStream(seqs, t0, t1, mixerLimit, sopts);
+    int token = audioEngine()->startStream(seqs, t0, t1, mixerLimit, sopts);
+    if (token != 0) {
+        ProjectAudioIO::Get(project).SetAudioIOToken(token);
+    }
+
+    return token > 0 ? muse::make_ok() : muse::make_ret(muse::Ret::Code::InternalError);
```

Comment:
maybe the comparison should be `token != 0` like above?
or even move the comparison to a separate variable and reuse it

Reply:
changed